### ISSUE TITLE
[Remote-SSH Bug]: Can not download vscode server for aarch64 Linux on commit 863d2581ecda6849923a2118d93a088b0745d9d6

### ISSUE THREAD

**Issue by emrainey** [microsoft/vscode-remote-release#10069]
### Is there an existing issue for this bug?

- [X] I have searched the existing issues

### Required Troubleshooting Steps

- [X] I have followed these troubleshooting steps
- [X] I have tried both values of the `remote.SSH.useLocalServer` setting

### Connect Locally

It connects successfully

### ->

```
[13:41:43.688] > ready: 0c3d0a563b90
[13:41:43.804] > Linux 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023
[13:41:43.805] Platform: linux
[13:41:43.921] > /usr/bin/zsh
[13:41:43.922] Shell: zsh
[13:41:43.922] Creating bash subshell inside "zsh"
[13:41:44.051] > bash version:  5.1.4(1)-release
[13:41:44.160] > bash version:  5.1.4(1)-release
> bash version:  5.1.4(1)-release
[13:41:44.269] > 0c3d0a563b90: running
[13:41:44.383] > Acquiring lock on /home/dev/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/vscode-remote-lock.dev.863d2581ecda6849923a2118d93a088b0745d9d6
> Installing to /home/dev/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6...
> 0c3d0a563b90%%1%%
> Downloading with wget
[13:41:45.875] > wget download failed
> https://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/server-linux-armhf/stable: 2024-07-16 13:40:45 ERROR 404: Not Found.
> printenv:
[13:41:45.879] >     SHELL=/usr/bin/zsh
>     LANGUAGE=en_US.UTF-8
>     PWD=/home/dev/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6
>     LOGNAME=dev
>     XDG_SESSION_TYPE=tty
>     MOTD_SHOWN=pam
>     HOME=/home/dev
>     LANG=en_US.UTF-8
>     VSCODE_AGENT_FOLDER=/home/dev/.vscode-server
>     SSH_CONNECTION=146.16.0.100 59580 146.16.0.4 22
>     XDG_SESSION_CLASS=user
>     USER=dev
>     SHLVL=1
>     XDG_SESSION_ID=10
>     XDG_RUNTIME_DIR=/run/user/1001
>     SSH_CLIENT=146.16.0.100 59580 22
>     LC_ALL=en_US.UTF-8
>     PATH=/usr/local/bin:/usr/bin:/bin:/usr/games
>     DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1001/bus
>     OLDPWD=/home/dev
>     _=/usr/bin/printenv
[13:41:45.989] > Trigger local server download
> 0c3d0a563b90:trigger_server_download
> artifact==server-linux-armhf==
> destFolder==/home/dev/.vscode-server/bin/==
> destFolder2==/863d2581ecda6849923a2118d93a088b0745d9d6/vscode-server.tar.gz==
> 0c3d0a563b90:trigger_server_download_end
> Waiting for client to transfer server archive...
> Waiting for /home/dev/.vscode-server/bin/863d2581ecda6849923a2118d93a088b0745d9d6/vscode-server.tar.gz.done and vscode-server.tar.gz to exist
>  
[13:41:45.991] Got request to download on client for {"artifact":"server-linux-armhf","destPath":"/home/dev/.vscode-server/bin//863d2581ecda6849923a2118d93a088b0745d9d6/vscode-server.tar.gz"}
[13:41:45.992] Downloading VS Code server locally...
[13:41:46.456] Terminating local server
[13:41:46.461] Resolver error: Error: Failed to download VS Code Server (Server returned 404)
```

### Expected Behavior

Connection to aarch64 server should just auto download the server. 

```
# On the server
$ uname -a
Linux uhilsim-1 6.1.21-v8+ #1642 SMP PREEMPT Mon Apr  3 17:24:16 BST 2023 aarch64 GNU/Linux
```

The download should 1) be for aarch64 not armhf and 2) should be found. 
 

### Steps To Reproduce

1. Normal VS Code Release
2. Connect to Host
3. Expect remote host to download server on commit
4. 404 failure.

### Remote-SSH Log

<details>
<summary>Remote-SSH Log</summary>
<p>

```
See above
```

</p>
</details>

### Anything else?

Local environment

> Version: 1.87.2 (Universal)
> Commit: 863d2581ecda6849923a2118d93a088b0745d9d6
> Date: 2024-03-08T15:21:31.043Z
> Electron: 27.3.2
> ElectronBuildId: 26836302
> Chromium: 118.0.5993.159
> Node.js: 18.17.1
> V8: 11.8.172.18-electron.0
> OS: Darwin arm64 23.5.0

```
# On the Server (aarch64 linux)
$ wget -O vscode-server.tar.gz --no-check-certificate http://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/server-linux-aarch64/latest
# also
$ wget -O vscode-server.tar.gz --no-check-certificate http://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/server-linux-armhf/latest
```
both use a proxy and return
```
URL transformed to HTTPS due to an HSTS policy
--2024-07-16 13:43:29--  https://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/server-linux-armhf/latest
Connecting to 146.16.0.100:3128... connected.
Proxy request sent, awaiting response... 404 Not Found
2024-07-16 13:43:29 ERROR 404: Not Found.
```
On another machine without a proxy (local mac)

```
$ wget -O vscode-server.tar.gz --no-check-certificate http://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/server-linux-aarch64/latest
URL transformed to HTTPS due to an HSTS policy
--2024-07-16 13:46:03--  https://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/server-linux-aarch64/latest
Resolving update.code.visualstudio.com (update.code.visualstudio.com)... 13.107.253.40
Connecting to update.code.visualstudio.com (update.code.visualstudio.com)|13.107.253.40|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-07-16 13:46:03 ERROR 404: Not Found.
```

**Comment by dokeabha** [comment]
I am also facing the same issue.

**Comment by roblourens** [comment]
I'm not sure why the link 404s, but this looks like a pretty old version of VS Code, can you try updating?

**Comment by spscream** [comment]
i have the same issue on x64 and ubuntu host. I got 404 on https://update.code.visualstudio.com/commit:863d2581ecda6849923a2118d93a088b0745d9d6/cli-alpine-x64/latest

**Comment by imatlopez** [comment]
I have the same issue connecting from macOS onto a windows 11 VM using openssh 9.5

I am able to fix it by reverting to remote-ssh 107.1, it breaks starting 108
